### DISCUSSION TITLE
discuss: Is standard APISIX dev workbench in need ?

### DISCUSSION THREAD

**Discussion by leslie-tsang** [apache/apisix#5178] [Ideas]
## Issue description
Hi, I found that the APISIX community does not have a standard `out-of-the-box` development environment kit so far.
For those who want to join in the APISIX community may have a hard time of launch a standard development environment. 

## Proposal

We need a repo to offer an `out-of-the-box` development environment. For Unix-like OS, container tech like `docker` seem to be a good shim layer for virtual dev runtime environment. For Windows-like OS, [`docker`](https://github.com/moby/moby) don't work as well as Unix-like OS (filesystem permission issue and etc), I found [`Vagrant`](https://www.vagrantup.com/) may slove this problem by provide a VM for docker.

This repo provide some friendly feature for developer :

* Unix-like OS
* [apisix-base Nginx](https://github.com/api7/apisix-build-tools)
* Unit test ready or CI
* Reproducible Dev ENV
* Infra As Code

## Solution

So I create a repo [`apisix-workbench`](https://github.com/NGX-AI/apisix-workbench) to work it out, right now apisix-workbench support several feature :

|feature|support|
| :---: | :----: |
|Unix-like OS|provide by [`Vagrant`](https://www.vagrantup.com/) or hosted OS|
|apisix-base Nginx|TODO|
|Unit test ready or CI|Testing on [`test`](https://github.com/NGX-AI/apisix-workbench/tree/test) branch|
|Reproducible Dev ENV|Support by Vagrantfile and provision script|
|Infra As Code|Support by Vagrantfile、docker-compose.yml and etc provision script|

Any feedback appricated. 🙏

**Comment by spacewander** [comment]
LGTM. Would you send the purpose to dev@apisix.apache.org? Thanks!

**Reply by leslie-tsang** [reply]
Good idea. Done.

**Comment by juzhiyuan** [comment]
😄
